odoo/addons/base/models/ir_demo.py

```diff
@@ -13,9 +13,15 @@ class IrDemo(models.TransientModel):
 
     @assert_log_admin_access
     def install_demo(self):
+        original_company_country_code = self.env.company.country_code
+
         force_demo(self.env)
+
+        targetCompany = self.env['res.company'].search([('country_code', '=', original_company_country_code)])
         return {
-            'type': 'ir.actions.act_url',
-            'target': 'self',
-            'url': '/web',
+            'type': 'ir.actions.client',
+            'tag': 'switch_company',
+            'params': {
+                'company': targetCompany.ids[0],
```

Comment:
rename to `target_company`, also you can do `target_company.id` directly since your search returns at most 1 record (is it possible that it doesn't return anything ? not sure that every l10n comes with a company)